packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/dev_server/proxy.rb

```diff
@@ -210,7 +210,7 @@ def get_response_headers(response, env)
         def request(method, path, headers: nil, query: [], form_data: nil, body_stream: nil)
           uri = URI.join("https://#{shop}", path)
 
-          if Environment.theme_access_password?
+          if Environment.theme_access_password? && sfr_can_respond_to?(path)
```

Comment:
@karreiro – Is it a viable solution? Or did we imagine something else with the Theme Access App?

Reply:
@Poitrin - it's viable solution. Still, I'd personally have a method called:

```ruby
def proxy_via_theme_access_app?
  return false unless Environment.theme_access_password?
  return false if path == "/localization"
  return false if path.start_with?("/cart/")

  true
end
```

Reply:
Thanks for the feedback, changed it!